src/libs/fsdrv/lv_fs_posix.c

```diff
@@ -327,11 +328,63 @@ static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p)
     int ret = closedir(dir_p);
     if(ret < 0) {
         LV_LOG_WARN("Could not close directory: errno: %d", errno);
-        return LV_FS_RES_FS_ERR;
+        return fs_errno_to_res(errno);
     }
 
     return LV_FS_RES_OK;
 }
+
+/**
+ * Convert an errno value to a lv_fs_res_t value
+ * @param errno_val an errno value
+ * @return a corresponding lv_fs_res_t value
+ */
+static lv_fs_res_t fs_errno_to_res(int errno_val)
+{
+    switch(errno_val) {
+        case 0:
+            return LV_FS_RES_OK;
+
+        case EIO: /* I/O error */
+            return LV_FS_RES_HW_ERR;
+
+        case EFAULT: /* Bad address */
+            return LV_FS_RES_FS_ERR;
+
+        case ENOENT: /* No such file or directory */
+            return LV_FS_RES_NOT_EX;
+
+        case ENOSPC: /* No space left on device */
+            return LV_FS_RES_FULL;
+
+        case EBUSY: /* Operation already in progress */
+            return LV_FS_RES_LOCKED;
```

Comment:
Why not busy to busy?
EAGAIN should be handled in lv_fs_posix.c by doing again.

Reply:
I don't think it's necessary. This should be left to the upper layer to do. `lv_fs_posix` only does pure API call passthrough.

Reply:
But no upper layer has done this. I remembered we have fixed it the `EAGAIN` issue for v8. It should be done in fs layer such as lv_fs_posix. Not all fs have EAGAIN.

I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

Reply:
According to the comment [here](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_fs.h#L40) `LV_FS_RES_LOCKED` means "The file is already opened". So mapping busy to "already opened" can be misleading. In light of that I also vote for busy to busy.

Reply:
> But no upper layer has done this. I remembered we have fixed it the `EAGAIN` issue for v8. It should be done in fs layer such as lv_fs_posix. Not all fs have EAGAIN.
> 
> I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

`EAGAIN` does not require a handle. This comment comes from the `write` function. It only needs to be processed when `O_NONBLOCK` is set. Our `fs_open` function does not set this flag.
```c
 *  EAGAIN
 *    Non-blocking I/O has been selected using O_NONBLOCK and the write
 *    would block.
```

Reply:
> I suggest remove EAGAIN and deal it internally, map EBUSY to EBUSY.

Done.